objects/tag.c

```diff
@@ -139,7 +139,6 @@ tag_view(lua_State *L, int udx, bool view)
     {
         tag->selected = view;
         banning_need_update();
-        ewmh_update_net_current_desktop();
```

Comment:
It depends on what for. First focused client will trigger the update, so actual clients depending on this atom should be fine. Pagers depend on NET_CURRENT_DESKTOP, but i doubt they worked correctly before with how it was handled.

This, of course, can be reverted, but then we also might need to update on mouse::enter events, to behave correctly.
Example: let's say we have chromium focused on one screen, then we switch to empty tag on second screen. Client on first screen is still focused, but CURRENT_DESKTOP now points to tag on the second screen. Now to move tabs we have to explicitely refocus chromium window (and with sloppy focus, for example, just moving mouse back to already focused window does not trigger focus event).

So we will need to update on screen change, on focus (to work properly with keyboard driven interactions), on tag switch AND on mouse enter event. Again i'm not sure about performance implications there.
No really good solutions, i'm afraid.